web/portal/brand/src/entities/Carrier/Action/BalanceOperations.tsx

```diff
@@ -0,0 +1,198 @@
+import { EntityValues } from '@irontec/ivoz-ui';
+import { MoreMenuItem } from '@irontec/ivoz-ui/components/List/Content/Shared/MoreChildEntityLinks';
+import { StyledTableRowCustomCta } from '@irontec/ivoz-ui/components/List/Content/Table/ContentTable.styles';
+import {
+  OutlinedButton,
+  SolidButton,
+} from '@irontec/ivoz-ui/components/shared/Button/Button.styles';
+import {
+  ActionFunctionComponent,
+  ActionItemProps,
+} from '@irontec/ivoz-ui/router/routeMapParser';
+import { StyledDropdown } from '@irontec/ivoz-ui/services/form/Field/Dropdown/Dropdown.styles';
+import { StyledTextField } from '@irontec/ivoz-ui/services/form/Field/TextField/TextField.styles';
+import _ from '@irontec/ivoz-ui/services/translations/translate';
+import ReplayIcon from '@mui/icons-material/Replay';
+import {
+  Box,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  SelectChangeEvent,
+  Tooltip,
+} from '@mui/material';
+import axios from 'axios';
+import { CurrencyPropertyList } from 'entities/Currency/CurrencyProperties';
+import { useEffect, useState } from 'react';
+import { useStoreActions } from 'store';
+
+import Carrier from '../Carrier';
+import { CarrierPropertyList } from '../CarrierProperties';
+
+const amountChoices = [
+  { id: '+', label: '+', operation: 'increment' },
+  { id: '-', label: '-', operation: 'decrement' },
+];
+
+const BalanceOperations: ActionFunctionComponent = (props: ActionItemProps) => {
+  const { row, variant = 'icon' } = props;
+  const [amountChoice, setAmountChoice] = useState(amountChoices[0].id);
+  const [amountValue, setAmountValue] = useState('0.00');
+  const [open, setOpen] = useState(false);
+  const [currencySymbol, setCurrencySymbol] = useState('');
+  const apiPost = useStoreActions((actions) => {
+    return actions.api.post;
+  });
+  const cancelTokenSource = axios.CancelToken.source();
+
+  const apiGet = useStoreActions((actions) => {
+    return actions.api.get;
+  });
+  const reload = useStoreActions((actions) => {
+    return actions.list.reload;
+  });
+  useEffect(() => {
+    apiGet({
+      path: `${Carrier.path}/${row.id}`,
+      params: {},
+      successCallback: async (value: CarrierPropertyList<EntityValues>) => {
+        const { symbol } = value.currency as CurrencyPropertyList<EntityValues>;
+        setCurrencySymbol(symbol);
+
+        return;
+      },
+      cancelToken: cancelTokenSource.token,
+    });
+  }, [currencySymbol, row.id]);
```

Comment:
currencySymbol or setCurrencySymbol?

Reply:
Isn't it the same? When currencySymbol observer changes, the useEffect will be dispatched again. Or is it preferable to use the function which sets the value of currencySymbol?

Reply:
If you use currencySymbol the effect will be triggered every time currencySymbol changes. You're probably getting something like: Empty value > Actual value > do request egain and get the same value.

If you use setCurrencySymbol it'll do one single request.